### PR TITLE
Update CAT to check for duplicate streams

### LINKED ISSUE #29481: New CAT test: validate all streams have unique names
From https://github.com/airbytehq/oncall/issues/2578 - if streams are duplicated (or rather, I believe, have the same name), the connector will succeed in the CAT test but will fail in the platform while running discover.

In the [fix](https://github.com/airbytehq/airbyte/pull/28709), we tested this via a unit test in the connector, but ideally CAT would not let this information through

### PR DESCRIPTION
**Pull request by jbfbell**
Per the oncall process request, I have added a test that checks for duplicate streams

### CODE REVIEW

File: airbyte-integrations/bases/connector-acceptance-test/connector_acceptance_test/tests/test_core.py [hunk 1]
```diff
@@ -652,6 +656,11 @@ async def test_discover(self, connector_config, docker_runner: ConnectorRunner):
         assert len(catalog_messages) == 1, "Catalog message should be emitted exactly once"
         assert catalog_messages[0].catalog, "Message should have catalog"
         assert catalog_messages[0].catalog.streams, "Catalog should contain streams"
+        assert self.streams_are_uniquely_named(catalog_messages[0].catalog.streams), "Catalog should have uniquely named streams"
+
+    def streams_are_uniquely_named(self, streams) -> bool:
+        names = [stream.name for stream in streams]
+        return len(set(names)) == len(names)
```
Comment (jbfbell) [inline airbyte-integrations/bases/connector-acceptance-test/connector_acceptance_test/tests/test_core.py:663]: This is the change that I made, not sure why all the format tagged along

Comment (bnchrch) [reply airbyte-integrations/bases/connector-acceptance-test/connector_acceptance_test/tests/test_core.py:663]: @jbfbell how did you run format?

Comment (jbfbell) [reply airbyte-integrations/bases/connector-acceptance-test/connector_acceptance_test/tests/test_core.py:663]: i did not 🤷 maybe it was intelliJ

File: airbyte-integrations/bases/connector-acceptance-test/connector_acceptance_test/tests/test_core.py [hunk 2]
```diff
@@ -652,6 +656,11 @@ async def test_discover(self, connector_config, docker_runner: ConnectorRunner):
         assert len(catalog_messages) == 1, "Catalog message should be emitted exactly once"
         assert catalog_messages[0].catalog, "Message should have catalog"
         assert catalog_messages[0].catalog.streams, "Catalog should contain streams"
+        assert self.streams_are_uniquely_named(catalog_messages[0].catalog.streams), "Catalog should have uniquely named streams"
```
Comment (bnchrch) [inline airbyte-integrations/bases/connector-acceptance-test/connector_acceptance_test/tests/test_core.py:655]: Can we update this to output which streams are duplicates?
```suggestion
        duplicate_streams = get_duplicates(catalog_messages[0].catalog.streams)
        assert len(duplicate_streams) == 0, f"Catalog should have uniquely named streams. Duplicate stream(s) found: {duplicate_streams}"
        
def get_duplicates(strings):
    counts = {string: strings.count(string) for string in strings}
    duplicates = [string for string, count in counts.items() if count > 1]
    return duplicates
```

File: airbyte-integrations/bases/connector-acceptance-test/connector_acceptance_test/tests/test_core.py (hunk 1 above)
Comment (bnchrch) [reply airbyte-integrations/bases/connector-acceptance-test/connector_acceptance_test/tests/test_core.py:659]: looks like the Auto formatter got it back inline.

However thats probably not a good configuration we have going on that caused the intellij format

**Review by bnchrch — APPROVED** [review]
One minor suggestion to improve, but other than that this seems fine to me.

Thanks Joe.

You rock.

**Review by alafanechere — COMMENTED** [review]
@jbfbell could you bump the CAT version in its pyproject.toml and Dockerfile - add an entry to `CHANGELOG.md` ?

**Pull request merged**